### PR TITLE
Update a remote value forwarding future to include an on clause

### PR DESCRIPTION
**Pull request by ronawho**
#4212 added a future to capture an rvf bug where we mistakenly forward
variables across memory barriers imposed by atomic operations. However, the
future only used task functions and #4220 is limiting rvf to only apply to on
clauses. This just updates the future so it still captures the bug by including
an on clause. It also throws --no-local since --local removes on clauses at
parse time.

### CODE REVIEW

**Comment by ronawho** [comment]
@noakesmichael look ok to you? Note that this will mean #4220 shouldn't remove the .future for this test anymore.

File: test/parallel/begin/deitz/test_begin_cobegin1_atomic.chpl [hunk 1]
```diff
@@ -9,8 +9,10 @@ proc main {
       s.write(1);
     }
     {
-      s.waitFor(1);;
-      z = x;
+      on Locales[0] {
```
Comment (noakesmichael) [inline test/parallel/begin/deitz/test_begin_cobegin1_atomic.chpl]: Hate to point it out but line 13 has two semi-colons.

Comment (ronawho) [inline test/parallel/begin/deitz/test_begin_cobegin1_atomic.chpl]: whoops.. I'll probably just let that go since it doesn't really hurt anything

Comment (ronawho) [inline test/parallel/begin/deitz/test_begin_cobegin1_atomic.chpl]: eh, decided to change it

**Pull request merged**